src/dmd/expressionsem.d

```diff
@@ -535,16 +535,16 @@ private Expression resolveUFCS(Scope* sc, CallExp ce)
                 ce.e1 = ey;
                 if (isDotOpDispatch(ey))
                 {
-                    uint errors = global.startGagging();
-                    e = ce.syntaxCopy().expressionSemantic(sc);
-                    if (!global.endGagging(errors))
-                        return e;
-
                     // even opDispatch and UFCS must have valid arguments,
                     // so now that we've seen indication of a problem,
                     // check them for issues.
                     Expressions* originalArguments = Expression.arraySyntaxCopy(ce.arguments);
```

Comment:
This copy is now created but discarded when `expressionSemantic` succeeds below.

Maybe a better fix would be to check whether the `e.arguments.` has any errors after the initial `expressionSemantic`. That should avoid the copy + still report invalid arguments?

Reply:
The problem is that expressionSemantic destructively modifies the AST. Prior to this patch, a syntax copy of the entire CallExp was made (and therefore, for the arguments also) regardless of whether semantic succeeded. Right now, only the arguments are copied. I don't see any way around it.